Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -138,29 +138,40 @@ private Branch(Context context) {
 		linkCache_ = new HashMap<BranchLinkData, String>();
 	}
 
-	@Deprecated
-	public static Branch getInstance(Context context, String key) {
+	public static Branch getInstance(Context context, String branchKey) {
 		if (branchReferral_ == null) {
 			branchReferral_ = Branch.initInstance(context);
 		}
 		branchReferral_.context_ = context;
-		branchReferral_.prefHelper_.setAppKey(key);
+        if (branchKey.startsWith("key_")) {
+            branchReferral_.prefHelper_.setBranchKey(branchKey);
+        } else {
+            branchReferral_.prefHelper_.setAppKey(branchKey);
+        }
 		return branchReferral_;
 	}
 
-	public static Branch getInstance(Context context) {
+	private static Branch getBranchInstance(Context context, boolean isLive) {
 		if (branchReferral_ == null) {
 			branchReferral_ = Branch.initInstance(context);
 
-			String appKey = branchReferral_.prefHelper_.getAppKey();
-	        if (appKey == null || appKey.equalsIgnoreCase(PrefHelper.NO_STRING_VALUE)) {
-	        	Log.i("BranchSDK", "Branch Warning: Please enter your Branch App Key in your project's res/values/strings.xml!");
+			String branchKey = branchReferral_.prefHelper_.getBranchKey(isLive);
+	        if (branchKey == null || branchKey.equalsIgnoreCase(PrefHelper.NO_STRING_VALUE)) {
```

Comment:
Well, the purpose of this warning is prompt partners to transition from using app key to branch key, as we will abandon app key eventually. Will make sure the sdk is pushed out after backend.